File: backend/src/main/java/com/ecommerce/backend/service/CategoryService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.entity.Category;

import java.util.List;

public interface CategoryService {

    Category createCategory(String name);

    List<Category> getAllCategories();

    Category getCategoryById(Long id);

    Category updateCategory(Long id, String newName);

    void deleteCategory(Long id);
}

File: Frontend/src/app/core/services/category.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Category } from "../models/category.model";

@Injectable({ providedIn: 'root' })
export class CategoryService {
  private baseUrl = 'http://localhost:8080/api/categories';

  constructor(private http: HttpClient) {}

  getAllCategories(): Observable<Category[]> {
    return this.http.get<Category[]>(this.baseUrl);
  }

  createCategory(name: string): Observable<Category> {
    return this.http.post<Category>(this.baseUrl, null, { params: { name } });
  }

  deleteCategory(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
